overlord/snapstate/snapstate.go

```diff
@@ -462,20 +462,66 @@ func UpdateMany(st *state.State, names []string, userID int) ([]string, []*state
 		}
 	}
 
-	updated := make([]string, 0, len(updates))
+	params := func(update *snap.Info) (string, Flags, *SnapState) {
+		snapst := stateByID[update.SnapID]
+		return snapst.Channel, snapst.Flags, snapst
+
+	}
+
+	return doUpdate(st, names, updates, params, userID)
+}
+
+func doUpdate(st *state.State, names []string, updates []*snap.Info, params func(*snap.Info) (channel string, flags Flags, snapst *SnapState), userID int) ([]string, []*state.TaskSet, error) {
 	tasksets := make([]*state.TaskSet, 0, len(updates))
+	var retiredAutoAliasesTs *state.TaskSet
+
+	var nameSet map[string]bool
+	if len(names) != 0 {
+		nameSet = make(map[string]bool, len(names))
+		for _, name := range names {
+			nameSet[name] = true
+		}
+	}
+
+	newAutoAliases, mustRetireAutoAliases, transferTargets, err := autoAliasesUpdate(st, names, updates)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	reportUpdated := make(map[string]bool, len(updates))
+
+	if len(mustRetireAutoAliases) != 0 {
```

Comment:
Just a general comment (sorry that I'm not more helpful). Makes me feel a bit uneasy that the alias handling has more code than the snap update handling. I wonder if it would make the code easier to read if we could move the "retire" and "new" aliases handling in helpers so that the doUpdate() code itself is a bit more compact? Both code blocks looks somewhat similar (maybe only at a superficial level though).